### PR TITLE
Change product update route to a patch request

### PR DESCRIPTION
**Pull request by b-tess**
## Description

Please include a summary of the changes and the related issue.
Add any screenshots if a change has been made on the client side.

Screenshot 2024-07-30 141219

Use setattr function to update a Product object

Slight change to the post route for products to include columns added to the model

Add logic to get and edit one product to the frontend

Display and edit one product on the frontend

Add input focus, on button click, when needing
to add a supply request on the products page

Add logic to post a new product on the frontend

Add the stores to the features in Redux

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by SFrasia — APPROVED** [review]
Looks good

File: client/src/features/products/productSlice.js [hunk 1]
```diff
@@ -18,9 +22,80 @@ export const getAllProducts = createAsyncThunk('products/getAll', async () => {
...
 
+//Get one product
+export const getOneProduct = createAsyncThunk(
+    'products/getOne',
+    async (productItemId, thunkAPI) => {
+        try {
+            // console.log(
+            //     'Get one product function called with id:',
+            //     productItemId
+            // )
```
Comment (otsembo) [inline client/src/features/products/productSlice.js:33]: Once you are done with logs. Remove them.

File: client/src/features/products/productSlice.js [hunk 2]
```diff
@@ -18,9 +22,80 @@ export const getAllProducts = createAsyncThunk('products/getAll', async () => {
...
+)
+
+//Edit one product
+export const editOneProduct = createAsyncThunk(
+    'products/editOne',
+    async ({ productItemId, updateData }, thunkAPI) => {
+        try {
+            // console.log('Updated product from slice:')
+            // console.log(productItemId)
+            // console.log(updateData)
```
Comment (otsembo) [inline client/src/features/products/productSlice.js:55]: Remove unnecessary logs

**Review by otsembo — APPROVED** [review]
Looks good. Check comments for a few pieces of feedback.

**Pull request merged**